### PR TITLE
feat(api): Add option to set urlSessionDelegate

### LINKED ISSUE #965: Support for SSL pinning
For security reasons our application lets the SentrySDK send all data via our own server.  As an extra safety measure we would like to use SSL pinning and TrustKit. Will this be supported in the future? It would suffice to be able to get access to the url session used by the SentrySDK. Thanks!

### PR DESCRIPTION
**Pull request by joptriodos**
Add urlSessionDelegate option to SentryOptions. This delegate is used for all network data-transfer tasks performed by Sentry. 

## :scroll: Description

Add one option to SentryOptions, "urlSessionDelegate", which is an NSURLSessionDelegate. If defined, this urlSessionDelegate is set as the delegate of the NSURLSession that is used by SentryClient for performing network data-transfer tasks. 

## :bulb: Motivation and Context

At Triodos for security reasons we need to be able to apply SSL pinning. This is made possible by being able to set the url session's delegate. 
Fixes #965

## :green_heart: How did you test it?

I added a unit test which validates that the NSURLSessionDelegate's delegate functions are called. 
At Triodos we currently make use of a fork of sentry-cocoa, in which the urlSessionDelegate option is available as well. We make use of this option to successfully apply SSL pinning. 

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the CHANGELOG if needed
- [x] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - feat: Add transaction to the scope #992
 - fix: Pass SentryTracer to span child #1040
 - feat: Add span to SentrySDK #1042
+- feat: Add urlSessionDelegate option to SentryOptions #965
```
Comment (brustolin) [inline CHANGELOG.md:15]:
```suggestion
- feat: Add urlSessionDelegate option to SentryOptions #1044
```

You can link the issue in this PR.
CHANGELOG entries need to link to the PR where things were changed/added.

**Review by brustolin — APPROVED** [review]
I like the ideia. 

Code looks good to me.

**Review by bruno-garcia — APPROVED** [review]
The change makes sense.

**Comment by bruno-garcia** [comment]
Thanks @joptriodos !

**Comment by joptriodos** [comment]
Great, thanks for reviewing!

**Comment by teameh** [comment]
Nice work! @joptriodos 👍🏼

**Pull request merged**